mlrun/api/main.py

```diff
@@ -471,6 +527,51 @@ def _cleanup_runtimes():
         close_session(db_session)
 
 
+async def _stop_logs():
+    """
+    Stop logs for runs that are in terminal state and last updated in the last hour
+    """
+    logger.debug(
+        "Getting all runs which reached terminal state in the past hour",
+    )
+    db_session = await fastapi.concurrency.run_in_threadpool(create_session)
+    runs = await fastapi.concurrency.run_in_threadpool(
+        get_db().list_distinct_runs_uids,
+        db_session,
+        requested_logs_modes=[True],
+        only_uids=False,
+        states=mlrun.runtimes.constants.RunStates.terminal_states(),
+        last_update_time_from=datetime.datetime.now(datetime.timezone.utc)
+        - datetime.timedelta(seconds=1.5 * config.log_collector.stop_logs_interval),
+    )
+
+    logger.debug(
+        "Stopping logs for runs which reached terminal state in the past hour",
+        runs_count=len(runs),
+    )
+    await _stop_logs_for_runs(runs)
+
+
+async def _stop_logs_for_runs(runs: list):
+    project_to_run_uids = {}
+    for run in runs:
+        project_name = run.get("metadata", {}).get("project", None)
+        run_uid = run.get("metadata", {}).get("uid", None)
+        project_to_run_uids.setdefault(project_name, []).append(run_uid)
+
+    if project_to_run_uids:
+        try:
+            await log_collector.LogCollectorClient().stop_logs(
+                project_to_run_uids_dict=project_to_run_uids,
+            )
+        except Exception as exc:
+            logger.warning(
+                "Failed stopping logs for run. Ignoring",
+                exc=err_to_str(exc),
+                project_to_run_uids=project_to_run_uids,
+            )
```

Comment:
seems like would never reach that as you always silent the errors received from the log collector.

Reply:
well now looking at it a bit deeper you might reach that as you might fail some how on parsing the `StopLogRequest` is that was your intention?